peer.go

```diff
@@ -1723,6 +1702,16 @@ out:
 			// relevant sub-systems and launching a goroutine to
 			// wait for close tx conf.
 			p.finalizeChanClosure(chanCloser)
+
+		// The channel reannounce delay has elapsed, broadcast the
+		// reenabled channel updates to the network. This should only
+		// fire once, so we set the dispatchStatuses channel to nil to
+		// mark it for garbage collection. If the peer is torn down
+		// before firing, reenabling will not be attempted.
+		case <-reenableTimeout:
+			p.reenableActiveChannels()
+			reenableTimeout = nil
```

Comment:
is nilling necessary?

Reply:
shaves some CPU cycles from select, figured why not?

Reply:
adds some CPU cycles in my head at least

Reply:
Why btw will it save CPU cycles? I think maybe that is only for tickers.

Reply:
because the runtime doesn't have to acquire a lock to check if a value is being sent on a nil channel, it just skips it. it saves CPU cycles on every subsequent select

Reply:
```
func main() {
        const n = 1000000
        const numChans = 10

        chans := make([]chan int, numChans)
        for i := range chans {
                chans[i] = make(chan int)
        }   

        start := time.Now()
        for i := 0; i < n; i++ {
                select {
                case <-chans[0]:
                case <-chans[1]:
                case <-chans[2]:
                case <-chans[3]:
                case <-chans[4]:
                case <-chans[5]:
                case <-chans[6]:
                case <-chans[6]:
                case <-chans[8]:
                case <-chans[9]:
                default:
                }   
        }   
        fmt.Printf("non-nil: %v\n", time.Since(start))

        for i := range chans {
                chans[i] = nil 
        }   

        start = time.Now()
        for i := 0; i < n; i++ {
                select {
                case <-chans[0]:
                case <-chans[1]:
                case <-chans[2]:
                case <-chans[3]:
                case <-chans[4]:
                case <-chans[5]:
                case <-chans[6]:
                case <-chans[6]:
                case <-chans[8]:
                case <-chans[9]:
                default:
                }   
        }   
        fmt.Printf("nil: %v\n", time.Since(start))
}
```
running this gives, OMM:
```
non-nil: 521.206479ms
nil: 155.758026ms
```

Reply:
In case anyone was interested, I modified the demo to show how the latency is linearly dependent on the number of nil channels:
```
func main() {
        const n = 1000000
        const numChans = 10

        chans := make([]chan int, numChans)
        for i := range chans {
                chans[i] = make(chan int)
        }   

        testNNil := func(ii int) {
                start := time.Now()
                for i := range chans[:ii] {
                        chans[i] = nil 
                }   

                start = time.Now()
                for i := 0; i < n; i++ {
                        select {
                        case <-chans[0]:
                        case <-chans[1]:
                        case <-chans[2]:
                        case <-chans[3]:
                        case <-chans[4]:
                        case <-chans[5]:
                        case <-chans[6]:
                        case <-chans[6]:
                        case <-chans[8]:
                        case <-chans[9]:
                        default:
                        }   
                }   
                fmt.Printf("%d-nil: %v\n", ii, time.Since(start))
        }   

        for i := 0; i < 11; i++ {
                testNNil(i)
        }   
}
```
outputs:
```
0-nil: 519.117008ms
1-nil: 505.169358ms
2-nil: 485.513103ms
3-nil: 456.82318ms
4-nil: 430.449548ms
5-nil: 387.429949ms
6-nil: 340.134363ms
7-nil: 236.290499ms
8-nil: 238.719919ms
9-nil: 191.990285ms
10-nil: 157.386867ms
```

Reply:
Wow, nice! TIL :)